nav2_costmap_2d/src/costmap_2d.cpp

```diff
@@ -184,6 +184,29 @@ bool Costmap2D::copyCostmapWindow(
   return true;
 }
 
+bool Costmap2D::copyWindow(
+  const Costmap2D & source,
+  unsigned int sx0, unsigned int sy0, unsigned int sxn, unsigned int syn,
+  unsigned int dx0, unsigned int dy0)
+{
+  const unsigned int sz_x = sxn - sx0;
+  const unsigned int sz_y = syn - sy0;
+
+  if (sxn > source.getSizeInCellsX() || syn > source.getSizeInCellsY()) {
```

Comment:
Was this a condition you ran into?

Reply:
I've met it in tests. However, there is no one is safe from incorrect input data. This prevents the program to go out of costmap sizes which could lead to a segfault crashes. Therefore I've added the verification of source boundaries the same as for destination boundaries as well.

Reply:
That's fine, just making sure, if you ran into this in actual situations, then it would make me question some other part of this PR that wasn't quite right. If it's just defensive, that's OK